### PR TITLE
Implement new feature: Gray-encoding/decoding

### PR DESCRIPTION
**Pull request by t-m-z**
For review and discussion:
The UI accepts
- decimal or binary input which could be encoded or decoded.
- The output will be the missing input format (binary or decimal) and the binary and decimal format of the encoded/decoded input.

I tested with with 
- the Wikipedia example and 
- a manually created example 99/1100011 <=> 82/10100010

Regarding tests - how do I manage the expected output?

### CODE REVIEW

File: lib/widgets/tools/crypto_and_encodings/gray.dart [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  String outputLine2 = '';
+  String outputLine3 = '';
+
+  var _decimalMaskFormatter = MaskTextInputFormatter(
+      mask: '#' * 10000, // allow 10000 characters input
+      filter: {"#": RegExp(r'[0-9]')}
+  );
+
+  var _binaryDigitsMaskFormatter = MaskTextInputFormatter(
+      mask: '#' * 5000, // allow 5000 4-digit binary blocks, spaces will be set automatically after each block
```
Comment (S-Man42) [inline lib/widgets/tools/crypto_and_encodings/gray.dart:38]: This mask doesn't fit the comment. This simply allows 5000 [01] characters. If you want to split them into blocks, you need '#### ' (four characters with a trailing space as block delimiter)

But I guess (from executing the tool) you don't want to divide them into blocks. So, please remove irritating comments :)

File: lib/widgets/tools/crypto_and_encodings/gray.dart [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        )
+    );
+
+    return
+      Column(
+      children: rows
+    );
+
+/*
+
```
Comment (S-Man42) [inline lib/widgets/tools/crypto_and_encodings/gray.dart:181]: Please remove the comments

File: assets/i18n/de.json [hunk 3]
```diff
@@ -873,6 +873,21 @@
   "gccode_example" : "GCK25B → 156997",
   "gccode_novalidgcode" : "Kein valider GC-Code",
 
+  "gray_title" : "Gray-Code",
+  "gray_description" : "Konvertiert eine Dezimalzahl in eine Dezimalzahl basierend auf der Gray-Codierung",
```
Comment (S-Man42) [inline assets/i18n/de.json:877]: In English you say decimal/binary. This should be done here as well :)

File: lib/widgets/registry.dart [hunk 4]
```diff
@@ -388,10 +389,16 @@ class Registry {
...
-        i18nPrefix: 'gronsfeld',
-        category: ToolCategory.CRYPTOGRAPHY,
-        searchStrings: SEARCHSTRING_VIGENERE + 'gronsfeld'
+          tool: Gray(),
+          i18nPrefix: 'gray',
+          category: ToolCategory.CRYPTOGRAPHY,
+          searchStrings: 'binary decimal gray binär dezimal enoding decoding verschlüsseln entschlüsseln'
+      ),
+      GCWToolWidget(
+          tool: Gronsfeld(),
```
Comment (S-Man42) [inline lib/widgets/registry.dart:398]: Please watch the indents :)

Comment (t-m-z) [reply lib/widgets/registry.dart:398]: Sorry - done. I have to keep an eye on this

File: lib/widgets/tools/crypto_and_encodings/gray.dart (hunk 1 above)
Comment (t-m-z) [reply lib/widgets/tools/crypto_and_encodings/gray.dart:38]: Done

File: lib/widgets/tools/crypto_and_encodings/gray.dart (hunk 2 above)
Comment (t-m-z) [reply lib/widgets/tools/crypto_and_encodings/gray.dart:181]: They will be removed in the final version of the UI.

File: assets/i18n/de.json (hunk 3 above)
Comment (t-m-z) [reply assets/i18n/de.json:877]: Done

**Comment by t-m-z** [comment]
checks for some issues are necessary

**Pull request closed without merging**